presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -901,6 +912,13 @@ private ConnectorMetadata getMetadataForWrite(Session session, ConnectorId conne
         return new JsonCodecFactory(provider).jsonCodec(ViewDefinition.class);
     }
 
+    private Predicate<TableLayoutResult> hasAllDesiredColumns(Optional<Set<ColumnHandle>> desiredColumns)
```

Comment:
this implementation is currently not correct, because `Optional.empty()` stands for "all columns" in this case. Unlike `Optional` of empty list which stands for "no columns".